events/moose/moose.go

```diff
@@ -306,19 +306,20 @@ func (s *Subscriber) NotifyConnect(data events.DataConnect) error {
 		return nil
 	}
 
-	s.connectedAt = time.Now()
-	dnsResolutionTime := int(data.DNSResolutionTime.Milliseconds())
 	var threatProtection moose.Enum_SS_NordvpnappOptBool
 	if data.ThreatProtectionLite {
 		threatProtection = moose.Enum_SS_NordvpnappOptBool(moose.OptBoolTrue)
 	} else {
 		threatProtection = moose.Enum_SS_NordvpnappOptBool(moose.OptBoolFalse)
 	}
 
+	eventDuration := int(time.Since(s.connectedAt).Seconds())
 	var result moose.Enum_SS_NordvpnappEventStatus
 	switch data.Type {
 	case events.ConnectAttempt:
 		result = moose.Enum_SS_NordvpnappEventStatus(moose.EventStatusAttempt)
+		s.connectedAt = time.Now()
```

Comment:
Maybe rename the variable to `s.connectionStart` or something like that? Because in fact this is the timestamp of connection attempt start, not the moment of connection success.